crates/aptos-rosetta/src/types/objects.rs

```diff
@@ -1811,6 +1826,25 @@ fn get_amount_from_event(events: &[ContractEvent], event_key: &EventKey) -> Vec<
     })
 }
 
+/// Filter v2 FeeStatement events with non-zero storage_fee_refund
+fn get_fee_statement_from_event(events: &[ContractEvent]) -> Vec<FeeStatement> {
+    events
+        .iter()
+        .filter(|event| event.is_v2())
+        .filter_map(|event| {
+            if let Ok(fee_statement_event) = bcs::from_bytes::<FeeStatement>(event.event_data()) {
+                if fee_statement_event.storage_fee_refund() != 0 {
+                    Some(fee_statement_event.clone()) // Collect only if storage_fee_refund_octas is non-zero
+                } else {
+                    None
+                }
```

Comment:
You may be able to do this as a reference, but the cloning should be fine for now.